### PR TITLE
Add copyright/license headers to script files

### LINKED ISSUE #5827: Add copyright/license headers to all scripts
In our `make lint` and `make fmt` targets we ensure that all Go source files have a copyright & license comments like this:
```sh
$ head -2 cmd/jaeger/main.go
// Copyright (c) 2023 The Jaeger Authors.
// SPDX-License-Identifier: Apache-2.0
```

Objective:
  * modify `./scripts/updateLicense.py` to also work for shell scripts under `./scripts/`
  * make sure not to override the shebang line
  * also make sure this script works for the Makefiles, which have the pattern of `Makefiles*` or `*.mk`
  * invoke the script as needed from `lint` and `fmt` targets

Verification:
  * running `make lint` should flag all the scripts and Makefiles that are missing the license headers
  * running `make lint` should update them as needed

DO NOT commit the changes to the file headers, only raise a PR containing changes to the script & make targets.

### PR DESCRIPTION
**Pull request by Zen-cronic**
## Which problem is this PR solving?
- Resolves #5827 

## Description of the changes
- Update the `update_go_license` function to handle the `.sh` files and makefiles
- Update the [ALL_SRC](https://github.com/jaegertracing/jaeger/blob/dc49b1faaeee15c2498c562309bab4fc0525bd56/Makefile#L32) variable in the main `Makefile` to include the new file types, which in turn updates the `lint` and `fmt` targets

## How was this change tested?
- 

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: scripts/updateLicense.py [hunk 1]
```diff
@@ -79,7 +95,21 @@ def main():
...
+        elif name.endswith('.sh'):
+            try:
+                update_license(name, LICENSE_BLOB_LINES_SHELL_AND_MAKEFILE)
+            except Exception as error:
+                logger.error('Failed to process file %s', name)
+                logger.exception(error)
+                raise error
+        elif 'Makefile' in name:
+            try:
+                update_license(name, LICENSE_BLOB_LINES_SHELL_AND_MAKEFILE)
```
Comment (Zen-cronic) [inline scripts/updateLicense.py:112]: Reminder: the blocks inside `if` and `elif`'s are somewhat repeated.

File: Makefile [hunk 2]
```diff
@@ -29,7 +29,8 @@ endif
 
 
 # All .go files that are not auto-generated and should be auto-formatted and linted.
-ALL_SRC = $(shell find . -name '*.go' \
+# .sh files and makefiles are included.
+ALL_SRC = $(shell find . \( -name '*.go' -o -name '*.sh' -o -name 'Makefile*' \) \
```
Comment (yurishkuro) [inline Makefile:33]: This var is meant for go sources. You can define another one for scripts and pass both to the license fixer script

File: scripts/updateLicense.py [hunk 3]
```diff
@@ -17,10 +17,15 @@
     ('// ' + l).strip() + '\n' for l in LICENSE_BLOB.split('\n')
 ]
 
+LICENSE_BLOB_LINES_SHELL_AND_MAKEFILE = [
```
Comment (yurishkuro) [inline scripts/updateLicense.py:20]: Change to a function that accepts comment prefix

File: scripts/updateLicense.py (hunk 1 above)
Comment (yurishkuro) [reply scripts/updateLicense.py:112]: You only need to determine comment prefix based on file type. Then try/catch can be done once.

File: scripts/updateLicense.py (hunk 3 above)
Comment (Zen-cronic) [reply scripts/updateLicense.py:20]: got it, i was thinking that too

File: Makefile [hunk 4]
```diff
@@ -43,6 +42,11 @@ ALL_SRC = $(shell find . \( -name '*.go' -o -name '*.sh' -o -name 'Makefile*' \)
 				   -type f | \
 				sort)
 
+# All .sh or .py or Makefile or .mk files that should be auto-formatted and linted.
+SCRIPTS_SRC = $(shell find . \( -name '*.sh' -o -name '*.py' -o -name 'Makefile*' \) \
```
Comment (Zen-cronic) [inline Makefile:46]: Also finds the files in submodules: `jaeger-ui`, `idl`, `idl/opentelemetry-proto`. Is that behaviour appropriate?

Comment (yurishkuro) [reply Makefile:46]: no, we don't want to go to submodules, need to exclude them.

File: scripts/updateLicense.py [hunk 5]
```diff
@@ -70,16 +91,34 @@ def replace(header_lines):
                 f.write(line)
         print(name)
 
+def get_license_type(file):
+    ext_map = {
+        '.go' : LICENSE_BLOB_LINES_GO,
+        '.mk' : LICENSE_BLOB_LINES_MAKEFILE,
+        'Makefile' : LICENSE_BLOB_LINES_MAKEFILE,
+        '.py' : LICENSE_BLOB_LINES_PYTHON,
+        '.sh' : LICENSE_BLOB_LINES_SHELL,
```
Comment (yurishkuro) [inline scripts/updateLicense.py:100]: why do you need the vars, can you not just call `get_license_blob_lines(...)` inline?

Comment (yurishkuro) [reply scripts/updateLicense.py:100]: at minimum you can define local vars, and you only need two of them (all scripts share the same)

Comment (Zen-cronic) [reply scripts/updateLicense.py:100]: > why do you need the vars, can you not just call `get_license_blob_lines(...)` inline?

i thought it'd be more readable

**Comment by yurishkuro** [comment]
Thanks!

**Pull request merged**